src/main/java/org/pac4j/vertx/handler/impl/RequiresAuthenticationHandler.java

```diff
@@ -193,15 +193,15 @@ protected void authorise(final User user, final RoutingContext context) {
         );
     }
 
-    protected boolean useSession(final WebContext context, final List<Client> currentClients) {
+    protected boolean useSession(final List<Client> currentClients) {
         return currentClients == null || currentClients.size() == 0 || currentClients.get(0) instanceof IndirectClient;
     }
 
-    protected boolean startAuthentication(final VertxWebContext context, final List<Client> currentClients) {
```

Comment:
The WebContext is included to enable override of the behaviour based on the request. I don't think we should remove it as a parameter from this method (any comment Jérôme?)